File: code/python/560.py
#https://leetcode.com/problems/subarray-sum-equals-k/
def subarraySum(self, nums: List[int], k: int) -> int:
    sum, res = 0, 0
    mp = {0: 1}
    for num in nums:
        sum += num
        if sum - k in mp:
            res += mp[sum - k]
        mp[sum] = mp.get(sum, 0) + 1
    return res

File: code/cpp/560.cpp
//https://leetcode.com/problems/subarray-sum-equals-k/
int subarraySum(vector<int>& nums, int k) {
    size_t sum = 0, res = 0;
    unordered_map<int, int> mp;
    mp[0] = 1;
    for (int n : nums) {
        sum += n;
        if (mp.contains(sum - k)) res += mp[sum - k];
        mp[sum]++;
    }
    return res;
}
